common/utils/utils.go

```diff
@@ -492,3 +495,91 @@ func DialectToTarget(dialect string) string {
 	}
 	return constants.TargetSpanner
 }
+
+// PopulateMigrationData populates migration data like source schema details,
+// request id, target dialect, connection mechanism etc in conv object
+func PopulateMigrationData(conv *internal.Conv, driver, targetDb string) {
+
+	migrationRequestId := "HB-" + uuid.New().String()
+
+	migrationData := migration.MigrationData{
+		MigrationRequestId: &migrationRequestId,
+	}
+	populateMigrationDataSourceDetails(driver, &migrationData)
+	populateMigrationDataSchemaPatterns(conv, &migrationData)
+
+	switch targetDb {
+	case constants.TargetSpanner:
+		migrationData.TargetDialect = migration.MigrationData_GOOGLE_STANDARD_SQL.Enum()
+	case constants.TargetExperimentalPostgres:
+		migrationData.TargetDialect = migration.MigrationData_POSTGRES.Enum()
+	}
+	conv.MigrationData = migrationData
+}
+
+// populateMigrationDataSchemaPatterns populates schema petterns like number of tables, foreign key, primary key,
+// indexes, interleaves, max interleave depth and if source schema is missing primary key in migrationData object
+func populateMigrationDataSchemaPatterns(conv *internal.Conv, migrationData *migration.MigrationData) {
+
+	numTables := int32(len(conv.SrcSchema))
+	var numForeignKey, numIndexes, numMissingPrimaryKey, numInterleaves, maxInterleaveDepth int32 = 0, 0, 0, 0, 0
+
+	for _, table := range conv.SrcSchema {
+		numForeignKey += int32(len(table.ForeignKeys))
+		if len(table.PrimaryKeys) == 0 {
+			numMissingPrimaryKey++
+		}
+		numIndexes += int32(len(table.Indexes))
+	}
+
+	for _, table := range conv.SpSchema {
+		depth := 0
+		parentTableName := table.Name
+		for conv.SpSchema[parentTableName].Parent != "" {
+			numInterleaves++
+			depth++
+			parentTableName = conv.SpSchema[parentTableName].Parent
+		}
+		maxInterleaveDepth = int32(math.Max(float64(maxInterleaveDepth), float64(depth)))
```

Comment:
Max operation is because the depth of all interleaves will be different and we want the max of all. And typecasting was done because math.Max takes float64 arguments.